include/my_stacktrace.h

```diff
@@ -59,6 +59,24 @@ void my_set_exception_pointers(EXCEPTION_POINTERS *ep);
 void my_write_core(int sig);
 #endif
 
+/* print stack traces for all threads */
+void my_pstack();
+
+/* print stack traces for all threads before assert failures */
+#define assert_with_stack_traces(expr) \
```

Comment:
It looks we'd end up with two sets of asserts/aborts - with stacktraces and without. Is there any reason not to use the stacktrace version uniformly? One thing I can think of is that my_pstack requires some stack space, and we might be aborting because stack is full. This can be checked though.